source/helpers/get-deinit-handler.ts

```diff
@@ -1,10 +1,14 @@
 export default function getDeinitHandler(deinit: Deinit): VoidFunction {
 	if (deinit instanceof MutationObserver || deinit instanceof ResizeObserver || deinit instanceof IntersectionObserver) {
-		return deinit.disconnect;
+		return () => {
+			deinit.disconnect();
+		};
 	}
 
 	if ('abort' in deinit) { // Selector observer
-		return deinit.abort;
+		return () => {
+			deinit.abort();
+		};
 	}
 
 	if ('destroy' in deinit) { // Delegate subscription
```

Comment:
@fregante Do you think `delegation.destroy` needs the same treatment?

Reply:
Not required in either:

https://github.com/fregante/delegate-it/blob/ec81f4400a6bc1d48e75705a6a8089f6cf6207c8/index.ts#L165-L170

https://github.com/sindresorhus/debounce-fn/blob/7234eb37aa74198dfca2ab49a889cb24555360dc/index.js#L69-L79